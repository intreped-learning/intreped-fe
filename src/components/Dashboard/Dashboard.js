import React from 'react';
import { Link } from 'react-router-dom';
import './Dashboard.scss';
import CourseCard from '../CourseCard/CourseCard';
import { useSelector } from 'react-redux';


const Dashboard = () => {
  const { courses, teacher } = useSelector(state => state);
  
  const filterCompleted = teacher.teacher_courses.filter(course => course.is_complete === true);

  const filterWatchList = teacher.teacher_courses.filter(course => course.is_complete === false);

  const getCourseInfo = (list) => {
    return list.map(course => courses.filter( e => e.id === course.course_id)[0]);
  }

  const formatCourseInfo = list => {
    return list.map(course => {
    const route = `courses/${course.id}`
      return (
        <Link to={route}>
          <div className='dashboard-course'>
            <h5 className='course-title'>{course.title}</h5>
            <img className='course-thumbnail' src={course.thumbnail} alt={course.title}/>
          </div>
        </Link>  
      )
    })
  }

  const watchlistCourses = getCourseInfo(filterWatchList)
  const completedCourses = getCourseInfo(filterCompleted)

  return (
    <main className='dashboard'>
      <h2>Completed Courses</h2>
      <h6>Click To Rewatch!</h6>
      <section className='courses-in-progress'>
        {formatCourseInfo(completedCourses)}
        {!completedCourses.length && <h2 className='blank-field-msg'>No Courses Started Yet!</h2>}
      </section>
      <h2>Watch List</h2>
      <section className='watch-list'>
        {formatCourseInfo(watchlistCourses)}
        {!watchlistCourses.length && <h2 className='blank-field-msg' >Add Some Courses To Your Watch List!</h2>}
      </section>
      <h2>Badges</h2>
      <section className='badges'>
        {!teacher.teacher_badges.length && <h2 className='blank-field-msg'>Complete Some Courses To Start Earning Badges!</h2>}
      </section>
    </main>
  )
}


export default Dashboard;
