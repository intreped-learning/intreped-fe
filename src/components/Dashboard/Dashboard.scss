.dashboard {
  background: #ebebeb;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;

  h2 {
    color: #2B3551;
  }

  section {
    border: 5px solid #2B3551;
    border-radius: 10px;
    display: flex;
    flex-direction: row;
    height: 40vh;
    overflow: scroll;
    padding: 10px;
    width: 90vw;
  }
}

.dashboard-course {
  background: #3b4a7f;
  border-radius: 10px;
  color: #ffffff;
  height: 35vh;
  padding: 4px;
  margin: 5px;
  width: 35vw;
}

.dashboard-course:hover {
  height: 36vh;
  width: 36vw;
}

.course-title {
  margin-top: 5px;
}

.course-thumbnail {
  height: 65%;
}

.blank-field-msg {
  align-self: center;
  margin: 0 auto;
}