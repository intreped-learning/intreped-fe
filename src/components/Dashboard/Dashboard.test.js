import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { BrowserRouter as Router} from 'react-router-dom';
import Dashboard from './Dashboard';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  teacher: {
    username: '',
    id: null,
    teacher_courses: [
      {
        "model": "intreped.course",
        "id": 1,
        "fields": {
          "title": "Kagan Cooperative Learning: Inside-Outside Circle & Talking Chips",
          "description": "Kagan structures are techniques and activities teachers can use in their classrooms to keep students engaged, active, and have fun while they are learning",
          "url": "https://www.youtube.com/watch?v=B-hTXzUTxf8",
          "thumbnail": "https://i.ytimg.com/vi/hTXzUTxf8/mqdefault.jpg",
          "category": "Engagement Strategies",
          "duration": "11m30s"
        }
      },
      {
        "model": "intreped.course",
        "id": 2,
        "fields": {
          "title": "Think Pair Share Explained",
          "description": "This video provides a succinct explanation for students about how to participate in THINK PAIR SHARE activities.",
          "url": "https://www.youtube.com/watch?v=wW87rihT38I",
          "thumbnail": "https://i.ytimg.com/vi/wW87rihT38I/mqdefault.jpg",
          "category": "Engagement Strategies",
          "duration": "1m35s"
        }
      }
    ],
    teacher_badges: []
  },
  courses: [
    {
      "model": "intreped.course",
      "id": 1,
      "fields": {
        "title": "Kagan Cooperative Learning: Inside-Outside Circle & Talking Chips",
        "description": "Kagan structures are techniques and activities teachers can use in their classrooms to keep students engaged, active, and have fun while they are learning",
        "url": "https://www.youtube.com/watch?v=B-hTXzUTxf8",
        "thumbnail": "https://i.ytimg.com/vi/hTXzUTxf8/mqdefault.jpg",
        "category": "Engagement Strategies",
        "duration": "11m30s"
      }
    },
    {
      "model": "intreped.course",
      "id": 2,
      "fields": {
        "title": "Think Pair Share Explained",
        "description": "This video provides a succinct explanation for students about how to participate in THINK PAIR SHARE activities.",
        "url": "https://www.youtube.com/watch?v=wW87rihT38I",
        "thumbnail": "https://i.ytimg.com/vi/wW87rihT38I/mqdefault.jpg",
        "category": "Engagement Strategies",
        "duration": "1m35s"
      }
    }
  ],
  modalOpen: true
});
const getWrapper = () => mount(
  <Router>
    <Provider store={store}>
      <Dashboard id={1}/>
    </Provider>
  </Router>
);
const wrapper = getWrapper();

describe('Dashboard', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});