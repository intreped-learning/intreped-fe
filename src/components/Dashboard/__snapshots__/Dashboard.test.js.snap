// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Dashboard
        id={1}
      >
        <main
          className="dashboard"
        >
          <h2>
            Completed Courses
          </h2>
          <h6>
            Click To Rewatch!
          </h6>
          <section
            className="courses-in-progress"
          >
            <h2
              className="blank-field-msg"
            >
              No Courses Started Yet!
            </h2>
          </section>
          <h2>
            Watch List
          </h2>
          <section
            className="watch-list"
          >
            <h2
              className="blank-field-msg"
            >
              Add Some Courses To Your Watch List!
            </h2>
          </section>
          <h2>
            Badges
          </h2>
          <section
            className="badges"
          >
            <h2
              className="blank-field-msg"
            >
              Complete Some Courses To Start Earning Badges!
            </h2>
          </section>
        </main>
      </Dashboard>
    </Provider>
  </Router>
</BrowserRouter>
`;
